src/FsAutoComplete.Core/Commands.fs

```diff
@@ -677,158 +633,273 @@ module Commands =
           Position = pos
           Scope = ic.ScopeKind }))
 
+  /// * `includeDeclarations`:
+  ///   if `false` only returns usage locations and excludes declarations
+  ///   * Note: if `true` you can still separate usages and declarations from each other
+  ///     with `Symbol.partitionInfoDeclarationsAndUsages`
+  /// * `includeBackticks`:
+  ///   if `true` returns ranges including existing backticks, otherwise without:
+  ///   `let _ = ``my value`` + 42`
+  ///   * `true`: ` ``my value`` `
+  ///   * `false`: `my value`
+  /// * `errorOnFailureToFixRange`:
+  ///   Ranges returned by FCS don't just span the actual identifier, but include Namespace, Module, Type: `System.String.IsNullOrEmpty`
+  ///   These ranges gets adjusted to include just the concrete identifier (`IsNullOrEmpty`)
+  ///   * If `false` and range cannot be adjust, the original range gets used.
+  ///     * When results are more important than always exact range
+  ///       -> for "Find All References"
+  ///   * If `true`: Instead of using the source range, this function instead returns an Error
+  ///     * When exact ranges are required
+  ///       -> for "Rename"
   let symbolUseWorkspace
-    getDeclarationLocation
-    (findReferencesForSymbolInFile: (string * FSharpProjectOptions * FSharpSymbol) -> Async<Range seq>)
+    (getDeclarationLocation: FSharpSymbolUse * NamedText -> SymbolDeclarationLocation option)
+    (findReferencesForSymbolInFile: (string<LocalPath> * FSharpProjectOptions * FSharpSymbol) -> Async<Range seq>)
     (tryGetFileSource: string<LocalPath> -> ResultOrString<NamedText>)
-    getProjectOptionsForFsproj
+    (tryGetProjectOptionsForFsproj: string<LocalPath> -> FSharpProjectOptions option)
+    (getAllProjectOptions: unit -> FSharpProjectOptions seq)
+    (includeDeclarations: bool)
+    (includeBackticks: bool)
+    (errorOnFailureToFixRange: bool)
     pos
     lineStr
     (text: NamedText)
     (tyRes: ParseAndCheckResults)
-    =
+    : Async<Result<(FSharpSymbol * IDictionary<string<LocalPath>, Range[]>), string>> =
```

Comment:
Prev this function returned declarations & usages separately (one Dictionary for decls, one for usages).  
I changed this to return all together in a single Dictionary: Everywhere the results of `symbolUseWorkspace` are used, these two are immediately thrown together again. This way `symbolUseWorkspace` as well as using its results is easier.  

If you don't need decls there's a parameter for that.  

And in case you want the results partitioned: Use `Symbol.partitionIntoDeclarationsAndUsages`